neuralpp/symbolic/sympy_expression.py

```diff
@@ -312,11 +338,63 @@ def from_sympy_function_and_general_arguments(sympy_function: sympy.Basic, argum
             # see test/sympy_test.py: test_sympy_bug()
             sympy_object = sympy_function(*[sympy_argument.sympy_object for sympy_argument in sympy_arguments],
                                           evaluate=False)
-            return SymPyFunctionApplication(sympy_object, type_dict)
+        elif sympy_function == sympy.Piecewise:
+            sympy_object = sympy_piecewise_from_if_then_else(
```

Comment:
How do we know `len(arguments) == 3` here? Can't we call this function with more arguments than 3?

Reply:
No. Because we are always converting from other *if-then-else* style conditional expressions (`functions.cond` or `z3.If()`. So the length of arguments must be 3.